cpp/src/arrow/flight/client.h

```diff
@@ -253,13 +264,22 @@ class ARROW_FLIGHT_EXPORT FlightClient {
   /// \param[in] options Per-RPC options
   /// \param[in] descriptor the dataset request, whether a named dataset or
   /// command
-  /// \param[out] schema_result the SchemaResult describing the dataset schema
-  /// \return Status
+  /// \return Arrow result with the SchemaResult describing the dataset schema
+  arrow::Result<std::unique_ptr<SchemaResult>> GetSchema(
```

Comment:
FYI, it would have been fine to delay these changes for another PR, but I appreciate it!